core/src/main/java/com/amplifyframework/core/model/query/predicate/BetweenQueryOperator.java

```diff
@@ -18,18 +18,19 @@
 import androidx.core.util.ObjectsCompat;
 
 /**
- * Represents a between condition with a starting and ending value for comparison.
+ * Represents an inclusive between condition with a starting and ending value for comparison.
+ * @param <T> Comparable data type of the field
  */
-public final class BetweenQueryOperator extends QueryOperator {
-    private Object start;
-    private Object end;
+public final class BetweenQueryOperator<T extends Comparable<T>> extends QueryOperator<T> {
+    private T start;
+    private T end;
 
     /**
      * Constructs a between condition.
      * @param start the value to be used for the beginning of the comparison range
      * @param end the value to be used for the end of the comparison range
      */
-    public BetweenQueryOperator(Object start, Object end) {
+    BetweenQueryOperator(T start, T end) {
```

Comment:
What's going on with null-ability in this class? Is it allowed for `start`, `end`, and the `field` argument to evaluate?

Reply:
These query operators will be evaluated by `QueryPredicateOperation`'s evaluate call, which will return `false` upon encountering an exception (except select few). `NullPointerException` thrown by these operators will just cause them to be evaluated as false in the end.